### PR TITLE
Feature/186 fixes for participant store

### PR DESCRIPTION
**Pull request by algattik**
Refactored participant store for extensibility.

Also removed some obsolete content around JSON file-based registry (test assets, docker-compose variables).

Relates to https://github.com/agera-edc/MinimumViableDataspace/issues/186

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -8,8 +8,6 @@ services:
         JVM_ARGS: "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005"
     environment:
       EDC_API_AUTH_KEY: ApiKeyDefaultValue
-      NODES_JSON_FILES_PREFIX: registry-
-      NODES_JSON_DIR: /resources/registry
     ports:
       - "8181:8181"
       - "5005:5005"
```
Comment (cpeeyush) [inline docker-compose.yml:13]: volumes can be removed as well.

Adding comment here it is not reflected in diff.

File: extensions/participant-store-memory/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+plugins {
+    `java-library`
+}
+
+val rsApi: String by project
```
Comment (cpeeyush) [inline extensions/participant-store-memory/build.gradle.kts:5]: unused.
```suggestion
```

**Review by cpeeyush — APPROVED** [review]
Approved with small suggestions.

**Comment by algattik** [comment]
Moved to new repo

**Pull request closed without merging**